src/main/java/de/tum/in/www1/artemis/security/annotations/enforceRoleInCourse/EnforceRoleInCourseAspect.java

```diff
@@ -1,72 +1,36 @@
-import static de.tum.in.www1.artemis.security.annotations.AnnotationUtils.getIdFromSignature;
 
-import java.util.Optional;
-
-import org.aspectj.lang.ProceedingJoinPoint;
-import org.aspectj.lang.annotation.Around;
 import org.aspectj.lang.annotation.Aspect;
 import org.aspectj.lang.annotation.Pointcut;
 import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Component;
 
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.security.annotations.EnforceRoleInResourceAspect;
 import de.tum.in.www1.artemis.service.AuthorizationCheckService;
 
 @Profile(PROFILE_CORE)
 @Component
 @Aspect
-public class EnforceRoleInCourseAspect {
-
-    private final AuthorizationCheckService authorizationCheckService;
+public class EnforceRoleInCourseAspect extends EnforceRoleInResourceAspect {
 
     public EnforceRoleInCourseAspect(AuthorizationCheckService authorizationCheckService) {
-        this.authorizationCheckService = authorizationCheckService;
+        super(authorizationCheckService, EnforceRoleInCourse.class, EnforceAtLeastStudentInCourse.class, EnforceAtLeastTutorInCourse.class, EnforceAtLeastEditorInCourse.class,
+                EnforceAtLeastInstructorInCourse.class);
     }
 
     /**
      * Pointcut around all methods or classes annotated with {@link EnforceRoleInCourse}.
      */
     @Pointcut("@within(EnforceRoleInCourse) || @annotation(EnforceRoleInCourse) || execution(@(@EnforceRoleInCourse *) * *(..))")
-    public void callAt() {
-    }
-
-    /**
-     * Aspect around all methods for which a role in course has been activated. Will check if the user has the required role in the course and only execute the underlying method if
-     * the user has the required role. Will otherwise return forbidden (as response entity)
-     *
-     * @param joinPoint Proceeding join point of the aspect
-     * @return The original return value of the called method, if all features are enabled, a forbidden response entity otherwise
-     * @throws Throwable If there was any error during method execution (both the aspect or the actual called method)
-     */
-    @Around(value = "callAt()", argNames = "joinPoint")
-    public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
-        final var annotation = getAnnotation(EnforceRoleInCourse.class, joinPoint).orElseThrow();
-        final var courseIdFieldName = getCourseIdFieldName(annotation, joinPoint);
-        final var courseId = getIdFromSignature(joinPoint, courseIdFieldName).orElseThrow(() -> new IllegalArgumentException(
-                "Method annotated with @EnforceRoleInCourse must have a parameter named " + annotation.courseIdFieldName() + " of type long/Long."));
-        authorizationCheckService.checkIsAtLeastRoleInCourseElseThrow(annotation.value(), courseId);
-        return joinPoint.proceed();
+    @Override
+    protected void callAt() {
```

Comment:
Is this intended to do nothing?

Reply:
Yes, this is just the point cut. For most purposes (including ours) this method should be empty.